src/main/core/scheduler/scheduler.c

```diff
@@ -585,7 +585,13 @@ void scheduler_awaitStart(Scheduler* scheduler) {
     /* each thread will boot their own hosts */
     _scheduler_startHosts(scheduler);
 
-    /* everyone is waiting for the next round to be ready */
+    if (scheduler->policyType != SP_SERIAL_GLOBAL) {
+        /* when everyone finishes starting the hosts, the main thread can prepare
+         * round 1 */
+        countdownlatch_countDownAwait(scheduler->finishBarrier);
```

Comment:
I was a bit confused both about reusing `finishBarrier` here, and why wait on `prepareRoundBarrier` below doesn't already accomplish what you want.

I see now that `prepareRoundBarrier` is actually waited at the *end* of a round in `scheduler_continueNextRound`. Would moving it to the *start* of a round solve the bug without having to (re)use another barrier here, as well as being more consistent with its name?

Reply:
I had first added a new barrier, but then decided to re-use the `finishBarrier` since it wasn't needed at the start. But I can see how that could lead to issues later, so I will go back to using a separate one.

The `prepareRoundBarrier` needs to be at the end, because the state that is updated while the workers are waiting at the barrier (i.e., the clock) is what the workers use to break out of their running while loop. If I moved `prepareRoundBarrier` to the beginning, then we would need an extra check and break statement---which is certainly possible but also more lines of code. (Perhaps `transitionRoundBarrier` is a better name.)

Ultimately, I'm not really trying to refactor this admittedly ugly code right now, just trying to minimize the changes needed to get to a correct state. That being said, I do want to rewrite the scheduler in rust (in the dev branch) at some point, because (i) I want to implement a continuous round scheduler to replace the current discrete round scheduler, and (ii) I think we can design a more efficient scheduler/scheduling policy now that we don't have to worry about migrating plugin thread-local storage anymore. But this is all for another month :)